tools/PI/DevHome.PI/ViewModels/WatsonPageViewModel.cs

```diff
@@ -58,6 +52,54 @@ public void UpdateTargetProcess(Process process)
         }
     }
 
+    public void Dispose()
+    {
+        _watsonHelper?.Dispose();
+        GC.SuppressFinalize(this);
+    }
+
+    public void AddNewEntry(DateTime timeGenerated, string moduleName, string executable, string eventGuid)
+    {
+        var newEntry = new WatsonReport(timeGenerated, moduleName, executable, eventGuid);
+        _dispatcher.TryEnqueue(() =>
+        {
+            ReportEntries.Add(newEntry, entry => entry.DateTimeGenerated);
+        });
+    }
+
+    public void UpdateEntry(string eventGuid, string watsonLog, string directoryPath)
```

Comment:
Lemme see if I understand what this method is doing.
1. Find an already reported event that has the same guid as `eventGuid`
2. If an event is found, replace the existing WatsonLog with the passed in WatsonLog.
3. If an event was found, read all files under DirectoryPath and append the contents of each file to WatsonReportFile.

Does that sounds right?  Sorry, it took me a few minutes to understand what this is doing.